mongo/collection.go

```diff
@@ -582,6 +582,72 @@ func (coll *Collection) Count(ctx context.Context, filter interface{},
 	return dispatch.Count(ctx, cmd, coll.client.topology, coll.readSelector)
 }
 
+// CountDocuments gets the number of documents matching the filter. A user can supply a
+// custom context to this method, or nil to default to context.Background().
+//
+// This method uses countDocumentsAggregatePipeline to turn the filter parameter and options
+// into aggregate pipeline.
+func (coll *Collection) CountDocuments(ctx context.Context, filter interface{},
+	opts ...countopt.Count) (int64, error) {
+
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	pipelineArr, err := countDocumentsAggregatePipeline(filter, opts...)
+	if err != nil {
+		return 0, err
+	}
+
+	countOpts, err := countopt.BundleCount(opts...).Unbundle(true)
+	if err != nil {
+		return 0, err
+	}
+
+	oldns := coll.namespace()
+	cmd := command.CountDocuments{
+		NS:          command.Namespace{DB: oldns.DB, Collection: oldns.Collection},
+		Pipeline:    pipelineArr,
+		Opts:        countOpts,
+		ReadPref:    coll.readPreference,
+		ReadConcern: coll.readConcern,
+	}
+	return dispatch.CountDocuments(ctx, cmd, coll.client.topology, coll.readSelector)
+}
+
+// EstimatedDocumentCount gets an estimate of the count of documents in a collection using collection metadata.
+func (coll *Collection) EstimatedDocumentCount(ctx context.Context,
+	opts ...countopt.EstimatedDocumentCount) (int64, error) {
+
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	var countOptions []countopt.Count
+	for _, opt := range opts {
+		switch t := opt.(type) {
+		case countopt.OptMaxTimeMs:
+			countOptions = append(countOptions, countopt.MaxTimeMs(int32(t)))
+		}
+	}
+
+	countOpts, err := countopt.BundleCount(countOptions...).Unbundle(true)
```

Comment:
Instead of doing the iteration above and changing the `[]countop.EstimatedDocumentCount` into  `[]countopt.Count`, let's just make it so `countopt.EstimatedDocumentCountBundle.Unbundle` return a `[]option.CountOptioner`, which we can directly provide to `command.Count`.